woocommerce/woocommerce
# Fix TreeControl bugs in single selection mode

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Fix a bug where items with children were not selected in single mode and fix a bug where navigation between items with tab was not working properly due to using a radio button.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open storybook using `pnpm --filter="storybook" storybook`
2. Go to TreeControl > Selection Single (http://localhost:6007/?path=/story/woocommerce-admin-experimental-treecontrol--selection-single)
3. Click one of the items that have children (Technology, Phones, etc.)
4. Check if it's value appears in the JSON (that means onSelect was called)
5. Go to Simple Tree (http://localhost:6007/?path=/story/woocommerce-admin-experimental-treecontrol--simple-tree)
6. Expand all nodes
7. Navigate through it using Tab and Shift + Tab.
8. Navigation should work and not skip any item.
